### PR TITLE
HDDS-7097. Container scanner log output lacks useful information

### PR DESCRIPTION
**Pull request by DaveTeng0**
## What changes were proposed in this pull request?
1. Add log for the exception object when container corruption happens to better identify the failure.
2. Remove the stack trace of the call to KeyValueContainer#markContainerUnhealthy (not very useful).

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-7097

## How was this patch tested?
Manually test in local cluster to verify the log message.

Example message format:
`ERROR keyvalue.KeyValueContainerCheck: Marking Container [{containerID}] UNHEALTHY as it failed metadata check. Exception: {exception stack trace}`

### CODE REVIEW

**Comment by DaveTeng0** [comment]
(cc. @errose28 )

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainerCheck.java [hunk 1]
```diff
@@ -401,6 +401,6 @@ private void handleCorruption(IOException e) {
     String errStr =
         "Corruption detected in container: [" + containerID + "] ";
     String logMessage = errStr + "Exception: [" + e.getMessage() + "]";
-    LOG.error(logMessage);
+    LOG.error(logMessage, e);
```
Comment (sodonnel) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainerCheck.java:404]: You could clean this up further by removing the two intermediate strings:

```
LOG.error("Corruption detected in container: [{}]. Exception: {}", containerID, e.getMessage, e);
```

Does logging the full stack trace via the last exception parameter not also log the e.getMessage() too?

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java [hunk 2]
```diff
@@ -338,7 +337,7 @@ public void markContainerUnhealthy() throws StorageContainerException {
     }
     LOG.warn("Moving container {} to state {} from state:{} Trace:{}",
             containerData.getContainerPath(), containerData.getState(),
-            prevState, StringUtils.getStackTrace(Thread.currentThread()));
+            prevState);
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java:340]: ` Trace:{}` is no longer needed.

**Review by adoroszlai — COMMENTED** [review]
> Manually test in local cluster to verify the log message.

Please provide example output.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java (hunk 2 above)
Comment (DaveTeng0) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java:340]: Ohh!! Sure yes it should be removed! Thanks for the check attila!

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainerCheck.java (hunk 1 above)
Comment (DaveTeng0) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainerCheck.java:404]: Oops yes it does contain e.getMessage() in the exception object! I'll remove the redundant e.getMessage()! Thanks for the check Stephen!

**Comment by DaveTeng0** [comment]
> > Manually test in local cluster to verify the log message.
> 
> Please provide example output.

Sure! I'll add in the PR description!

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainerCheck.java [hunk 3]
```diff
@@ -398,9 +396,7 @@ private void loadContainerData() throws IOException {
   }
 
   private void handleCorruption(IOException e) {
-    String errStr =
-        "Corruption detected in container: [" + containerID + "] ";
-    String logMessage = errStr + "Exception: [" + e.getMessage() + "]";
-    LOG.error(logMessage);
+    LOG.error("Marking Container [{}] UNHEALTHY as it failed metadata check." +
+                    " Exception: {}", containerID, e);
```
Comment (errose28) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainerCheck.java:400]:
```suggestion
    LOG.error("Corruption detected in container [{}]. Marking it UNHEALTHY.", containerID, e);
```

Technically the metadata or the data (the actual blocks) may have been corrupted when this method is called. The exception should tell us which one it is. Also we can let the logger handle the exception printing directly without needing to pass it as a string format arg.

Comment (DaveTeng0) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainerCheck.java:400]: ohh i see~ ok! Yeah will update the string format!!

**Review by errose28 — APPROVED** [review]
Thanks for the fix @DaveTeng0

**Pull request merged**